pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/juniperkits/JuniperKitCreationStatusTest.java

```diff
@@ -295,30 +296,33 @@ public void invalidKitTypeTest() {
         String participantId = "OHSALK_";
         int rand = new Random().nextInt() & Integer.MAX_VALUE;
         String kitType = "BLOOD";
-        List<KitRequestShipping> oldkits = KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", kitType);
-        String json = "{ \"firstName\":\"P\","
-                + "\"lastName\":\"T\","
-                + "\"street1\":\"415 Main st\","
-                + "\"street2\":null,"
-                + "\"city\":\"Cambridge\","
-                + "\"state\":\"MA\","
-                + "\"postalCode\":\"02142\","
-                + "\"country\":\"USA\","
-                + "\"phoneNumber\":\" 111 - 222 - 3344\","
-                + "\"juniperKitId\":\"JuniperTestKitId_" + rand + "\","
-                + "\"juniperParticipantID\":\"" + participantId + rand + "\","
-                + "\"skipAddressValidation\":false,"
-                + "\"juniperStudyID\":\"Juniper-test-guid\"}";
-
-        JuniperKitRequest juniperTestKit = new Gson().fromJson(json, JuniperKitRequest.class);
-        createdKitIds.add(juniperTestKit.getJuniperKitId());
-        KitResponse kitResponse =
-                nonPepperKitCreationService.createNonPepperKit(juniperTestKit, kitType, mockEasyPostUtil, ddpInstance);
-        Assert.assertEquals(KitResponse.ErrorMessage.UNKNOWN_KIT_TYPE, kitResponse.getErrorMessage());
-        Assert.assertEquals(kitResponse.getValue(), kitType);
-        List<KitRequestShipping> newKits = KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", kitType);
-        Assert.assertEquals(newKits.size(), oldkits.size());
-
+        try {
+            KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", kitType);
+        } catch (DsmInternalError e) {
+            Assert.assertTrue(e.getMessage().contains(String.format("No kit was found for study %s with kit type name %s", instanceName, kitType)));
+            String json = "{ \"firstName\":\"P\","
```

Comment:
I'm not sure what's going on here. In response to an error, the test creates a new Juniper Kit?

Reply:
This is an old test, but yes, I first try to get BLOOD kits for a study that is not configured to have one, so DSM throws an error for it. So here I make sure that the assumption that each study gets to have only one kit type with a certain name is getting checked here. After that I try to create a kit with that type that is not set for this study and I check that the DSM response to that is an ERROR of  `UNKNOWN_KIT_TYPE`